doc/examples/jupyter_notebook_prototyping/jupyter_notebook_prototyping_tutorial.rst

```diff
@@ -0,0 +1,352 @@
+
+        # config file libraries
+        from moveit_configs_utils import MoveItConfigsBuilder
+        from ament_index_python.packages import get_package_share_directory
+
+Once we have imported the required packages we need to define our ``moveit_py`` node configuration. We do this through using the ``MoveItConfigsBuilder`` as follows: ::
+
+        moveit_config = (
+                MoveItConfigsBuilder(robot_name="panda", package_name="moveit_resources_panda_moveit_config")
+                .robot_description(file_path="config/panda.urdf.xacro")
+                .trajectory_execution(file_path="config/gripper_moveit_controllers.yaml")
+                .moveit_cpp(
+                file_path=get_package_share_directory("moveit2_tutorials")
+                + "/config/jupyter_notebook_prototyping.yaml"
+                )
+                .to_moveit_configs()
+        ).to_dict()
+
+where we convert the generated configuration instance to a dictionary so we can use it to initialise our ``moveit_py`` node. Finally we initialise a ``moveit_py`` node: ::
+
+        # instantiate moveit_py instance and a planning component for the panda_arm
+        moveit = MoveItPy(node_name="moveit_py", config_dict=moveit_config)
+        panda_arm = moveit.get_planning_component("panda_arm")
+
+Motion Planning Example
+-----------------------
+We can start by demonstrating the planning and execution of a simple motion from within the notebook: ::
+
+        # set plan start state using predefined state
+        panda_arm.set_start_state("ready")
+
+        # set pose goal using predefined state
+        panda_arm.set_goal(goal_state_name = "extended")
+
+        # plan to goal
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                robot_trajectory = plan_result.trajectory
+                moveit.execute(robot_trajectory, blocking=True, controllers=[])
+
+We can perform motion planning interactively (see the motion planning tutorial for further details of the motion planning API). Suppose we are developing our code and we make a mistake such as follows: ::
+
+        # set plan start state using predefined state
+        panda_arm.set_start_state("ready") # This conflicts with the current robot configuration and will cause an error
+
+        # set goal using a pose message this time
+        pose_goal = PoseStamped()
+        pose_goal.header.frame_id = "panda_link0"
+        pose_goal.pose.orientation.w = 1.0
+        pose_goal.pose.position.x = 0.28
+        pose_goal.pose.position.y = -0.2
+        pose_goal.pose.position.z = 0.5
+        panda_arm.set_goal(goal_pose_msg = pose_goal, link_name = "panda_link8")
+
+        # plan to goal
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                robot_trajectory = plan_result.trajectory
+                moveit.execute(robot_trajectory, blocking=True, controllers=[])
```

Comment:
All the comments here need to match the updated code with the helper function and other changes